<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1056</width>
    <height>548</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>DSV 0.4.1</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow#MainWindow{background:rgb(142,142,142);}
QToolButton{background:rgb(142,142,142);}</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <property name="iconSize">
   <size>
    <width>128</width>
    <height>128</height>
   </size>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QWidget" name="widget" native="true">
      <property name="maximumSize">
       <size>
        <width>120</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="focusPolicy">
       <enum>Qt::StrongFocus</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="current_tool">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16</height>
          </size>
         </property>
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="view_btn">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>←</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="line_btn">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Polyline</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="rect_btn">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Rectangle</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="circle_btn">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Circle</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QPushButton" name="curve_btn">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Curve</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="curve_sbx">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>5</number>
           </property>
           <property name="value">
            <number>3</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="connect_btn">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Connect</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="combinate_btn">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Combinate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="split_btn">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Split</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="to_all_layers">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>To All Layers</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="rotate_btn">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Rotate</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="rotate_angle">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="suffix">
            <string>°</string>
           </property>
           <property name="minimum">
            <number>-360</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="flipx_btn">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Flip X</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="flipy_btn">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Flip Y</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1056</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="file_menu">
    <property name="title">
     <string>file</string>
    </property>
    <addaction name="open_file"/>
    <addaction name="save_file"/>
    <addaction name="saveas_file"/>
    <addaction name="auto_save"/>
   </widget>
   <widget class="QMenu" name="menusetting">
    <property name="title">
     <string>setting</string>
    </property>
    <addaction name="show_origin"/>
    <addaction name="auto_layering"/>
    <addaction name="remember_file_type"/>
    <addaction name="auto_aligning"/>
    <addaction name="actionadvanced"/>
   </widget>
   <addaction name="file_menu"/>
   <addaction name="menusetting"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <action name="open_file">
   <property name="text">
    <string>open</string>
   </property>
  </action>
  <action name="save_file">
   <property name="text">
    <string>save</string>
   </property>
  </action>
  <action name="saveas_file">
   <property name="text">
    <string>save as</string>
   </property>
  </action>
  <action name="auto_save">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>auto save</string>
   </property>
  </action>
  <action name="auto_layering">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>auto layering</string>
   </property>
  </action>
  <action name="remember_file_type">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>remeber file type</string>
   </property>
  </action>
  <action name="auto_aligning">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>auto aligning</string>
   </property>
  </action>
  <action name="actionadvanced">
   <property name="text">
    <string>advanced</string>
   </property>
  </action>
  <action name="show_origin">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>show origin</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>open_file</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>open_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>599</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save_file</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>save_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>599</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveas_file</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveas_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>599</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>auto_save</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>auto_save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>599</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotate_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>rotate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>501</x>
     <y>289</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>flipx_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>flip_x()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>501</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>flipy_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>flip_y()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>554</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>open_file()</slot>
  <slot>save_file()</slot>
  <slot>saveas_file()</slot>
  <slot>auto_save()</slot>
  <slot>rotate()</slot>
  <slot>flip_x()</slot>
  <slot>flip_y()</slot>
 </slots>
</ui>
